### ISSUE TITLE
False positive in Style/HashEachMethods

### ISSUE THREAD

**Issue by styx** [rubocop/rubocop#9839]
I have code like this:

```ruby
        ::ApplicationRecord.connection.execute(sql).values.each do |field|
          p field
        end
```
--------

## Expected behavior

No error is raised by Style/HashEachMethods. There is no such method `each_value` as it's not a Hash.

## Actual behavior
```
dates.rb:56:59: C: [Correctable] Style/HashEachMethods: Use each_value instead of values.each.
        ::ApplicationRecord.connection.execute(build_sql).values.each do |field|
```

## Steps to reproduce the problem

Run RuboCop on the code snippet.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.15.0 (using Parser 3.0.1.1, rubocop-ast 1.6.0, running on ruby 3.0.1 x86_64-linux)
  - rubocop-rails 2.9.1
```

**Comment by koic** [comment]
`Style/HashEachMethods` cop is already marked as unsafe. I've open #9840 that can use `AllowedReceivers` to rule out false positives.
